src/erlcloud_cur.erl

```diff
@@ -0,0 +1,133 @@
+-module(erlcloud_cur).
+
+
+%% AWS Cost and Usage Report API implementation
+%% https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/api-reference.html
+
+-include("erlcloud_aws.hrl").
+
+%% Initialization API
+-export([new/2, new/3, new/4, new/5]).
+
+%% API
+-export([
+    describe_report_definitions/1,
+    describe_report_definitions/2
+]).
+
+
+%%------------------------------------------------------------------------------
+%% Initialization API
+%%------------------------------------------------------------------------------
+
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                cur_port          = Port,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer(), string()) ->
+    aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port, Scheme) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                cur_port          = Port,
+                cur_scheme        = Scheme,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+%% @doc
+%% Cost & Usage Report API:
+%% https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/describe-report-definitions.html
+%%
+%% `
+%% {ok, R} = erlcloud_cur:describe_report_definitions(Config).
+%% `
+%%
+-spec describe_report_definitions(aws_config()) -> {ok, map()} | {error, any()}.
```

Comment:
current output:

```
{ok,#{<<"ReportDefinitions">> =>
          [#{<<"AdditionalArtifacts">> => [],
             <<"AdditionalSchemaElements">> => [],
             <<"Compression">> => <<"GZIP">>,
             <<"Format">> => <<"textORcsv">>,
             <<"ReportName">> => <<"test-report">>,
             <<"S3Bucket">> => <<"user-test">>,
             <<"S3Prefix">> => <<"prefix">>,
             <<"S3Region">> => <<"us-east-1">>,
             <<"TimeUnit">> => <<"DAILY">>},
           #{<<"AdditionalArtifacts">> => [],
             <<"AdditionalSchemaElements">> => [],
             <<"Compression">> => <<"GZIP">>,
             <<"Format">> => <<"textORcsv">>,
             <<"ReportName">> => <<"test-report3">>,
             <<"S3Bucket">> => <<"user-test">>,<<"S3Prefix">> => <<>>,
             <<"S3Region">> => <<"us-east-1">>,
             <<"TimeUnit">> => <<"DAILY">>}
         ]}}.
```

which is basically map version of the json which is return by AWS. 

Shall we leave it like that or possible do list of maps like:

```
{ok,[#{<<"AdditionalArtifacts">> => [],
             <<"AdditionalSchemaElements">> => [],
             <<"Compression">> => <<"GZIP">>,
             <<"Format">> => <<"textORcsv">>,
             <<"ReportName">> => <<"test-report">>,
             <<"S3Bucket">> => <<"user-test">>,
             <<"S3Prefix">> => <<"prefix">>,
             <<"S3Region">> => <<"us-east-1">>,
             <<"TimeUnit">> => <<"DAILY">>},
           #{<<"AdditionalArtifacts">> => [],
             <<"AdditionalSchemaElements">> => [],
             <<"Compression">> => <<"GZIP">>,
             <<"Format">> => <<"textORcsv">>,
             <<"ReportName">> => <<"test-report3">>,
             <<"S3Bucket">> => <<"user-test">>,<<"S3Prefix">> => <<>>,
             <<"S3Region">> => <<"us-east-1">>,
             <<"TimeUnit">> => <<"DAILY">>}
         ]}.
```

Reply:
I'm ok with current map output because AWS may add more properties at the same level with `ReportDefinitions`

Reply:
no they will not

Reply:
> Returns one or more ReportDefinition objects.

from AWS docs. yes they wrap it in one more `ReportDefinitions`. this given API will note get paginated. so i'll remove unnecessary wrapping